samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -19,51 +19,116 @@
 
 package org.apache.samza.container.grouper.stream;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.I0Itec.zkclient.ZkClient;
+import org.apache.commons.lang3.Validate;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
 import org.apache.samza.config.JobConfig;
+import org.apache.samza.config.JobCoordinatorConfig;
+import org.apache.samza.config.ZkConfig;
 import org.apache.samza.container.TaskName;
+import org.apache.samza.standalone.PassthroughJobCoordinatorFactory;
 import org.apache.samza.system.SystemStreamPartition;
+import org.apache.samza.zk.ZkJobCoordinatorFactory;
+import org.apache.samza.zk.ZkKeyBuilder;
+
+import static org.apache.samza.zk.ZkCoordinationUtilsFactory.*;
+import static org.apache.samza.zk.ZkJobCoordinatorFactory.*;
 
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
 
 /**
- * AllSspToSingleTaskGrouper, as the name suggests, assigns all partitions to be consumed by a single TaskInstance
- * This is useful, in case of using load-balanced consumers like the new Kafka consumer, Samza doesn't control the
- * partitions being consumed by a task. Hence, it is assumed that there is only 1 task that processes all messages,
- * irrespective of which partition it belongs to.
- * This also implies that container and tasks are synonymous when this grouper is used. Taskname(s) has to be globally
- * unique within a given job.
+ * AllSspToSingleTaskGrouper creates TaskInstances equal to the number of containers and assigns all partitions to be
+ * consumed by each TaskInstance. This is useful, in case of using load-balanced consumers like the high-level Kafka
+ * consumer and Kinesis consumer, where Samza doesn't control the partitions being consumed by the task.
  *
- * Note: This grouper does not take in broadcast streams yet.
```

Comment:
I have added validation for broadcast streams in the grouper factory. It is up to the individual job coordinators to see if they can support getProcessorNames API, otherwise throw an exception. Is there a way to validate if the consumers are load-balanced ? On the flip side, do we need to state that this is supported ONLY for load-balanced consumers ? This could be used with non-LB consumers as well and it does what it exactly states.